dynacred/src/Config.ts

```diff
@@ -0,0 +1,52 @@
+import { Roster } from "@dedis/cothority/network/proto";
+
+import toml from "toml";
+
+type ID = Buffer;
+
+export class Config {
+    static fromTOML(raw: string): Config {
+        const parsed = toml.parse(raw);
+
+        const parseIDField = (name: string, field: any): ID => {
+            if (typeof field !== "string") {
+                throw Error(`field "${name}" is not string`);
+            }
+            if (!(/[a-f0-9]{64}/).test(field)) {
+                throw Error(`field "${name}" is not of correct format`);
+            }
+
+            return Buffer.from(field, "hex");
+        };
+
+        const tryGetIDField = (name: string): ID | null => {
+            if (!(name in parsed)) {
+                return null;
+            }
+            return parseIDField(name, parsed[name]);
+        };
+
+        const getIDField = (name: string): ID => {
+            if (!(name in parsed)) {
+                throw Error(`field "${name}" not found in config`);
+            }
+            return parseIDField(name, parsed[name]);
+        };
+
+        return new Config(
+            getIDField("ByzCoinID"),
+            Roster.fromTOML(raw),
```

Comment:
I'm a little bit hesitant if it wouldn't be better to put the roster in it's own toml-table:

```
ByzCoinID = "..."

[roster]
  [[servers]]
    ...
  [[servers]]
```

Not sure if it's worth it. Any thoughts?

Reply:
I don't think it is worth it, the roster is already contained inside the `servers` key.